cou/utils/openstack.py

```diff
@@ -342,3 +360,74 @@ def is_charm_supported(cls, charm: str) -> bool:
         if not cls._OPENSTACK_LOOKUP:
             cls._OPENSTACK_LOOKUP = cls._generate_lookup(cls._DEFAULT_CSV_FILE)
         return cls._OPENSTACK_LOOKUP.get(charm, False)
+
+
+class AuxiliaryTrackMapping:
+    """Tracks for the Auxiliary OpenStack Charms projects."""
+
+    # NOTE (gabrielcocenza) map based on the following tables:
+    # https://docs.openstack.org/charm-guide/latest/project/charm-delivery.html
+
+    @classmethod
+    def generate_map(cls) -> defaultdict:
+        """Generate a map that helps to track the right channels for auxiliary charms.
+
+        :return: Dictionary containing the auxiliary services by Ubuntu series and
+            Openstack releases. E.g:
+            {
+                "focal":{
+                    "ceph":{
+                        "ussuri":"octopus",
+                        "victoria":"octopus",
+                        "wallaby":"pacific",
+                        "xena":"pacific",
+                        "yoga":"quincy"
+                    },
+                },
+                "jammy":{
+                    "ceph":{
+                        "yoga":"quincy",
+                        "zed":"quincy",
+                        "2023.1":"quincy"
+                    },
+                },
+            }
+        :rtype: defaultdict
+        """
+        ubuntu_series = ["focal", "jammy"]
+        track_to_openstack_mapping: defaultdict[Any, Any] = defaultdict(lambda: defaultdict(str))
+        for series in ubuntu_series:
```

Comment:
Too nested and hard to understand.
Easiest way is to separate file reading from parsing.
Since we will not have much items in these csv files, reading into memory and parsing into model makes it more readable.

Reply:
I agree with Erhan here.

Reply:
From the stand up, we come in the conclusion that is better to write a global variable to this.